### PR TITLE
fix: #725 Allow arquillian descriptor properties prefixed with env. a…

### PR DESCRIPTION
**Pull request by iocanel**
…s env variables to setup/teardown scripts in k8s/openshift.

#### Changes proposed in this pull request:

-
-
-

**Fixes**: #

### CODE REVIEW

File: kubernetes/kubernetes/src/test/java/org/arquillian/cube/kubernetes/impl/DefaultConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by iocanel on 6/7/17.
```
Comment (bartoszmajsak) [inline kubernetes/kubernetes/src/test/java/org/arquillian/cube/kubernetes/impl/DefaultConfigurationTest.java]: Do we need this header? :)

File: kubernetes/kubernetes/src/test/java/org/arquillian/cube/kubernetes/impl/DefaultConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        map.put("KEY_TWO", "VALUE2");
+        map.put("env.KEY3", "VALUE3");
+        map.put("env.key4", "VALUE4");
+        map.put("env.KEY_5", "VALUE5");
+
+        Map<String, String> result = DefaultConfiguration.filterEnvironmentVariables(map);
+        Assert.assertEquals(3, result.size());
+        Assert.assertTrue(result.containsKey("KEY3"));
+        Assert.assertTrue(result.containsKey("key4"));
+        Assert.assertTrue(result.containsKey("KEY_5"));
```
Comment (bartoszmajsak) [inline kubernetes/kubernetes/src/test/java/org/arquillian/cube/kubernetes/impl/DefaultConfigurationTest.java]: I think we are using `assertj`, so maybe worth a small change?

**Comment by iocanel** [comment]
@bartoszmajsak: addressed both comments

**Comment by lordofthejars** [comment]
Thank you so much, ready for squash then?

**Comment by bartoszmajsak** [comment]
Thanks @iocanel. Just wanted to ask you one thing. In `arquillian-core` we support expressions in `arquillian.xml` which refer to environment variables, e.g. `${env.JAVA_HOME}` will be evaluated to `JAVA_HOME` value. Do we need this extra filtering?

**Comment by lordofthejars** [comment]
`arquillian.xml` supports out of the box setting environemnt variables on it. For example:

```
<property name="home">${env.JAVA_HOME}</property>
```

Is this what you need?

**Comment by iocanel** [comment]
No, I don't need something like that. 

I am starting to this that this is a strong indication that this is going to wrong direction (since its too confusing)...

So let's start fresh:

 "How can I use a map property via arquillian.xml"?

Can I just use something like:

    <property key="scriptEnv">
        key1=value1
        key2=value2
        key3=value3
    </property>


and If so how can I read it?

**Comment by lordofthejars** [comment]
You will read this as String. So what you'll need to do is `new ByteInputStream(content)` and pass it to `Properties`.

**Comment by lordofthejars** [comment]
Notice that you can still use `${}` syntax anywhere in the string since Arquillian will just replace any occurrences on given configured string.

**Comment by iocanel** [comment]
Thanks I will update the pull request accordingly.

**Comment by lordofthejars** [comment]
Thank you so much.

**Comment by lordofthejars** [comment]
https://github.com/arquillian/arquillian-cube/issues/783

**Pull request merged**